microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/smallrye/reactive/messaging/main/module.xml

```diff
@@ -58,5 +58,7 @@
         <module name="org.reactivestreams"/>
         <module name="org.slf4j"/>
         <module name="org.wildfly.reactive.messaging.config" services="import"/>
+        <module name="io.smallrye.metrics"/>
+        <module name="org.eclipse.microprofile.metrics.api"/>
```

Comment:
Unless I've missed something, this a class visibility only dependency then, with no config requirement (i.e. there's no requirement to add the MP Metrics extensions/subsystem.)

If so that's good. Adding libs to the server footprint isn't great, but requiring adding endpoint-exposing subsystems to the config is yuck.

Reply:
This is the class https://github.com/smallrye/smallrye-reactive-messaging/blob/main/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/metrics/MetricDecorator.java which deals with it. It relies on a class from the API jar but in the absence of a CDI provided MetricsRegistry becomes a noop.

I have modified the io.smallrye.metrics dependency to be optional and verified my QS works for the following three scenarios:
* A standalone-microprofile.xml server. This has the microprofile-metrics subsystem, and its MetricsRegistry is injected
* A standalone.xml server. This does not have the microprofile-metrics subsystem, so no MetricsRegistry is injected. This works fine both:
    * out of the box
    * with the io.smallrye.metrics module deleted

Reply:
Thanks.